### PR TITLE
Barbarians raid #2588

### PR DESCRIPTION
**Pull request by bob0bob**
I changed how barbarians spawn.  You now have two values in the config file that can alter how barbarians spawn.  
This enhancement only deals with number of bars. spawning.

maxBarbarianSize = 6 - 40
spawnBarbarianSize = 1 -  10
Max will control the max number of bars. that can spawn at one time no matter that the spawn size is.
Spawn size is a value that controls how many bars. will spawn. The game will figure out the normal amount of bars. to spawn and the spawn size is a percentage  1 - 10 which represent 10 - 100%.  So if you only want 50% of the bars. spawning based on what the game wants then put a 5.

This has nothing to go with the difficulty. That is unchanged.
This way you have have hard, with few bars.  or easy two tons of bars.

### CODE REVIEW

File: src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java [hunk 1]
```diff
@@ -96,9 +96,10 @@
     public static final ResourceLocation BARBARIAN                    = new ResourceLocation(Constants.MOD_ID, "Barbarian");
     public static final ResourceLocation ARCHER                       = new ResourceLocation(Constants.MOD_ID, "ArcherBarbarian");
     public static final ResourceLocation CHIEF                        = new ResourceLocation(Constants.MOD_ID, "ChiefBarbarian");
-    public static final int              MAX_SIZE                     = Configurations.gameplay.maxBarbarianHordeSize;
-    public static final double           BARBARIANS_MULTIPLIER        = 0.5;
-    public static final double           ARCHER_BARBARIANS_MULTIPLIER = 0.25;
+    public static final double           BARBARIAN_SPAWN_SIZE         = ((double) Configurations.gameplay.spawnBarbarianSize * 0.1);
```
Comment (Raycoms) [inline src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java]: Always complicated to calc statics like that, if the player changes the config from the ingame editor it wouldn't reload this variable then.

Comment (Raycoms) [reply src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java]: I know the code was like that before, but ti would be nice to use the Configurations at the places directly, they are constants already basically.

**Comment by bob0bob** [comment]
Ok I made the change to use the variable directly.

File: src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java (hunk 1 above)
Comment (bob0bob) [reply src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java]: Done

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/2692/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

File: src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java [hunk 2]
```diff
@@ -96,9 +96,9 @@
     public static final ResourceLocation BARBARIAN                    = new ResourceLocation(Constants.MOD_ID, "Barbarian");
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java]: ### Errors:
- Field has no usages.

File: src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java [hunk 3]
```diff
@@ -96,9 +96,9 @@
...
     public static final ResourceLocation ARCHER                       = new ResourceLocation(Constants.MOD_ID, "ArcherBarbarian");
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java]: ### Errors:
- Field has no usages.

File: src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java [hunk 4]
```diff
@@ -96,9 +96,9 @@
...
     public static final ResourceLocation CHIEF                        = new ResourceLocation(Constants.MOD_ID, "ChiefBarbarian");
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java]: ### Errors:
- Field has no usages.

File: src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java [hunk 5]
```diff
@@ -96,9 +96,9 @@
...
-    public static final int              MAX_SIZE                     = Configurations.gameplay.maxBarbarianHordeSize;
-    public static final double           BARBARIANS_MULTIPLIER        = 0.5;
-    public static final double           ARCHER_BARBARIANS_MULTIPLIER = 0.25;
+    public static final int              BARBARIAN_MAX_SIZE           = Configurations.gameplay.maxBarbarianSize;
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java]: ### Errors:
- Field has no usages.

File: src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java [hunk 6]
```diff
@@ -96,9 +96,9 @@
...
+    public static final double           BARBARIANS_MULTIPLIER        = 0.6;
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java]: ### Errors:
- Field has no usages.

File: src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java [hunk 7]
```diff
@@ -96,9 +96,9 @@
...
+    public static final double           ARCHER_BARBARIANS_MULTIPLIER = 0.30;
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java]: ### Errors:
- Field has no usages.

File: src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java [hunk 8]
```diff
@@ -96,9 +96,9 @@
...
     public static final double           CHIEF_BARBARIANS_MULTIPLIER  = 0.1;
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java]: ### Errors:
- Field has no usages.

File: src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java [hunk 9]
```diff
@@ -96,9 +96,9 @@
...
     public static final int              PREFERRED_MAX_HORDE_SIZE     = 40;
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java]: ### Errors:
- Field has no usages.

File: src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java [hunk 10]
```diff
@@ -96,9 +96,9 @@
...
     public static final int              PREFERRED_MAX_BARBARIANS     = 22;
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java]: ### Errors:
- Field has no usages.

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 38233 |  1  |  5  |  -4  |
| Errors |  592  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=13763&tab=Inspection&buildTypeId=bt88)

**Comment by Raycoms** [comment]
There are some conflicts with your other pr.

**Comment by bob0bob** [comment]
Is there anything this one needs?   I appears to be all corrected?  let me know.

**Pull request merged**